Framework bits of hybrid CG-DG work

lindsayad opened the pull request:
From pull request #23986. Refs issue #24055

lindsayad committed:
Framework bits of hybrid CG-DG work

From pull request #23986. Refs issue #24055

moosebuild commented:
Job [OpenMPI](https://civet.inl.gov/job/1725480/) on d86dfca : invalidated by @lindsayad

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/1725440/) on d86dfca wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/25308/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/1725483/) on d86dfca wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>a5148c</th>
      <th colspan=4>#25308 d86dfc</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.65%</td>
      <td>85.66%</td>
      <td>+0.01%</td>
      <td>95.79%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>90166</td>
      <td>90262</td>
      <td>+96</td>
      <td>91</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15109</td>
      <td>15108</td>
      <td>-1</td>
      <td>4</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/25308/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/25308/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/scalar_transport)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/25308/coverage/xfem)

</details>



This comment will be updated on new commits.

GiudGiud commented on framework/src/bcs/ADConservativeAdvectionBC.C:
```diff
@@ -0,0 +1,75 @@
+      "to be determined implicitly (e.g. we don't have a Dirichlet condition)");
+  params.addParam<FunctionName>("velocity_function",
+                                "Function describing the values of velocity on the boundary.");
+  params.addClassDescription("DG for convection");
```
```suggestion
  params.addClassDescription("Boundary condition for conservative advection with discontinuous Galerkin");
```

GiudGiud commented on framework/src/dgkernels/ADDGConvection.C:
```diff
@@ -0,0 +1,69 @@
+{
+  InputParameters params = ADDGKernel::validParams();
+  params.addRequiredParam<MaterialPropertyName>("velocity", "Velocity vector");
+  params.addClassDescription("DG for convection");
```
```suggestion
  params.addClassDescription("Kernel for conservative advection using discontinuous Galerkin");
```

lindsayad committed:
Change material name in framework

lindsayad committed:
Address Guillaume review

- Fix typos
- Remove unnecessary include
- Better class descriptions
- Remove unnecessary design specification

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

lindsayad committed:
ADDGConvection -> ADDGAdvection

GiudGiud reviewed (approved):
looks good
